bex.go

```diff
@@ -29,19 +29,21 @@ type BexLPPriceProvider struct {
 	lpTokenAddress common.Address
```

Comment:
It would be nice to add an explanation where where you don't pass block to the eth context.